File: OP_PROJECT_JAVA/src/game/Animals/Antelope.java
package game.Animals;

import game.Animal;
import game.Position;
import game.World;

import java.util.Random;

public class Antelope extends Animal {
    private static final int ANTELOPE_STRENGTH = 4;
    private static final int ANTELOPE_INIT = 4;
    private static final String ANTELOPE_TYPE = "Antelope";


    public Antelope(Position position, World world) {
        super(ANTELOPE_TYPE, ANTELOPE_STRENGTH, ANTELOPE_INIT, position, world);
    }

    @Override
    protected boolean checkEscaped() {
        double escapeChance = new Random().nextDouble();
        // If the generated number is less than 0.5, Antelope escapes
        if (escapeChance < 0.5) {
            world.displayInformation("Antelope escaped from the fight!");
            return true;
        }
        return false;
    }

    @Override
    public void action() {
        // Generate a random direction for movement
        int direction = (int) (Math.random() * 4);
        Position newPosition = new Position(position.getY(), position.getX());

        switch (direction) {
            case 0: // Move up
                newPosition.setY(newPosition.getY() - 2);
                break;
            case 1: // Move down
                newPosition.setY(newPosition.getY() + 2);
                break;
            case 2: // Move left
                newPosition.setX(newPosition.getX() - 2);
                break;
            case 3: // Move right
                newPosition.setX(newPosition.getX() + 2);
                break;
            default:
                break;
        }

        if (world.isPositionWithinBounds(newPosition)) {
            if (world.getOrganismAtPosition(newPosition) != null) {
                collision(world.getOrganismAtPosition(newPosition));
            } else {
                moveOrganism(newPosition);
            }
        }
    }
}

File: OP_PROJECT_PYTHON/src/game/Animals/Antelope.py
from random import randint, random
from src.game.Animal import Animal
from src.game.Position import Position


class Antelope(Animal):
    ANTELOPE_STRENGTH = 4
    ANTELOPE_INIT = 4
    ANTELOPE_TYPE = "Antelope"

    def __init__(self, position, world):
        super().__init__(Antelope.ANTELOPE_TYPE, Antelope.ANTELOPE_STRENGTH, Antelope.ANTELOPE_INIT, position, world)

    def check_escaped(self):
        escape_chance = random()
        if escape_chance < 0.5:
            self.world.display_information("Antelope escaped from the fight!")
            return True
        return False

    def action(self):
        direction = randint(0, 3)
        new_position = Position(self.position.get_y(), self.position.get_x())

        if direction == 0:  # Move up
            new_position.set_y(new_position.get_y() - 2)
        elif direction == 1:  # Move down
            new_position.set_y(new_position.get_y() + 2)
        elif direction == 2:  # Move left
            new_position.set_x(new_position.get_x() - 2)
        elif direction == 3:  # Move right
            new_position.set_x(new_position.get_x() + 2)

        if self.world.is_position_within_bounds(new_position):
            organism_at_new_position = self.world.get_organism_at_position(new_position)
            if organism_at_new_position is not None:
                self.collision(organism_at_new_position)
            else:
                self.move(new_position)
